pkg/api/types.go

```diff
@@ -1803,6 +1811,15 @@ type SecretType string
 
 const (
 	SecretTypeOpaque SecretType = "Opaque" // Default; arbitrary user-defined data
+
+	// SecretTypeDockerConfig contains a dockercfg file that follows the same format rules as ~/.dockercfg
+	//
+	// Required fields:
+	// - Secret.Data["dockercfg"] - a serialized ~/.dockercfg file
```

Comment:
it might clarify things if you put 'DockerConfigKey' instead of "dockercfg" as the key, since it will establish the linkage between the two constants.